src/Controls/tests/TestCases.HostApp/Issues/Issue25224.xaml.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+
+namespace Maui.Controls.Sample.Issues
+{
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+	[Issue(IssueTracker.Github, 25224, "CollectionView - EmptyView with EmptyViewTemplate for Data template selector page throws an exception", PlatformAffected.iOS)]
+	public partial class Issue25224: ContentPage
```

Comment:
The build is failing:


```
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(58,19): error CS8618: Non-nullable property 'Details' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.19041.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(30,24): error CS8618: Non-nullable property 'DefaultTemplate' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.19041.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(31,24): error CS8618: Non-nullable property 'OtherTemplate' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.19041.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(93,45): error CS8612: Nullability of reference types in type of 'event PropertyChangedEventHandler EmptyViewWithDataTemplateSelectorViewModel.PropertyChanged' doesn't match implicitly implemented member 'event PropertyChangedEventHandler? INotifyPropertyChanged.PropertyChanged'. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(95,78): error CS8625: Cannot convert null literal to non-nullable reference type. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(46,11): error CS8618: Non-nullable property 'Monkeys' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(46,11): error CS8618: Non-nullable event 'PropertyChanged' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the event as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(54,19): error CS8618: Non-nullable property 'Name' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(56,19): error CS8618: Non-nullable property 'Location' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(58,19): error CS8618: Non-nullable property 'Details' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(30,24): error CS8618: Non-nullable property 'DefaultTemplate' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Pages\Controls\CollectionViewGalleries\EmptyViewGalleries\EmptyViewWithDataTemplateSelector.xaml.cs(31,24): error CS8618: Non-nullable property 'OtherTemplate' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable. [D:\a\_work\1\s\src\Controls\samples\Controls.Sample\Maui.Controls.Sample.csproj::TargetFramework=net9.0-windows10.0.20348.0]
    6126 Warning(s)
    63 Error(s)
```

Could you take a look?

Reply:
@jsuarezruiz , I’ve resolved the build issues mentioned above. Let me know if there’s anything else needed!